zproject/backends.py

```diff
@@ -658,10 +659,16 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
             chosen_email = backend.strategy.request_data().get('email')
 
         if not chosen_email:
+            avatars = {}  # Dict[str, str]
+            for email in verified_emails:
+                existing_account = get_user(email, realm)
+                avatars[email] = avatar_url(existing_account)
+
             return render(backend.strategy.request, 'zerver/social_auth_select_email.html', context = {
                 'primary_email': verified_emails[0],
                 'verified_non_primary_emails': verified_emails[1:],
-                'backend': 'github'
+                'backend': 'github',
+                'avatar_urls': avatars,
```

Comment:
I rewrote this commit to use a better approach for fetching users (`get_user_profile_by_email`, which doesn't take a `realm` argument, isn't supposed to be used outside tests, and I think we can just use a common structure for both sets of emails).